msal/src/main/java/com/microsoft/identity/client/internal/CommandParametersAdapter.java

```diff
@@ -360,8 +362,8 @@ public static SignUpStartCommandParameters createSignUpStartCommandParameters(
                 .challengeType(configuration.getChallengeTypes())
                 .userAttributes(userAttributes)
                 // Start of the flow, so there is no correlation ID to use from a previous API response.
-                // Set it to a default value.
-                .correlationId(DiagnosticContext.INSTANCE.getThreadCorrelationId())
+                // If default value "UNSET", generate a new UUID as correlationId.
+                .correlationId(Objects.equals(DiagnosticContext.INSTANCE.getThreadCorrelationId(), "UNSET") ? UUID.randomUUID().toString() : DiagnosticContext.INSTANCE.getThreadCorrelationId())
```

Comment:
This logic doesn't belong here. it's the responsibility of DiagnosticContext to generate an ID, and do additional null checks to ensure an actual ID is returned.

Reply:
Update the logic inside DiagnosticContext getThreadCorrelationId()